### PR TITLE
Add display_avatar_url property for Member & User

### PR DESCRIPTION
**Pull request by HyperGH**
### Summary
It is somewhat tedious to get an `avatar_url` that is guaranteed to always exist, this property aims to rectify this issue, and simplify getting avatars somewhat.

### Checklist
- [x] I have run `nox` and all the pipelines have passed.
- [ ] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/guilds.py [hunk 1]
```diff
@@ -424,6 +424,10 @@ def guild_avatar_url(self) -> typing.Optional[files.URL]:
     @property
     def default_avatar_url(self) -> files.URL:
         return self.user.default_avatar_url
+    
+    @property
+    def display_avatar_url(self) -> files.URL:
+        return self.guild_avatar_url or self.avatar_url or self.default_avatar_url
```
Comment (FasterSpeeding) [inline hikari/guilds.py:430]:
```suggestion
        return self.guild_avatar_url or super().display_avatar_url
```

Would help avoid duplication here

**Review by davfsa — CHANGES_REQUESTED** [review]
Thank you for the PR. Could you please add tests and the change log fragment file for this change?

Additionally, could you also add a `display_name` attribute? Should be similar to this one in both `User` and `Member`

If you have any questions on how to do that, please check CONTRIBUTING.md or ask us :)

File: hikari/users.py [hunk 2]
```diff
@@ -511,6 +511,11 @@ def default_avatar_url(self) -> files.URL:
             discriminator=int(self.discriminator) % 5,
             file_format="png",
         )
+    
+    @property
+    def display_avatar_url(self) -> files.URL:
+        """Display avatar URL for this user."""
+        return self.avatar_url or self.display_avatar_url
```
Comment (FasterSpeeding) [inline hikari/users.py:518]:
```suggestion
        return self.avatar_url or self.default_avatar_url
```

Is probably what you actually wanted here

**Review by FasterSpeeding — CHANGES_REQUESTED** [review]

**Comment by HyperGH** [comment]
> Thank you for the PR. Could you please add tests and the change log fragment file for this change?
> 
> If you have any questions on how to do that, please check CONTRIBUTING.md or ask us :)

Done!

**Review by davfsa — DISMISSED** [review]

File: hikari/users.py [hunk 3]
```diff
@@ -512,6 +512,11 @@ def default_avatar_url(self) -> files.URL:
             file_format="png",
         )
 
+    @property
+    def display_avatar_url(self) -> files.URL:
+        """Display avatar URL for this user."""
+        return self.avatar_url or self.default_avatar_url
```
Comment (FasterSpeeding) [inline hikari/users.py:518]: Missing test coverage

Comment (davfsa) [inline hikari/users.py:518]: Deleted the wrong comment while doing my review by mistake (*Original comment by snab*):

"Missing test coverage"

Comment (davfsa) [inline hikari/users.py:518]: Would it be worth avoiding a call less here and just doing

```suggestion
        return self.make_avatar_url() or self.default_avatar_url
```

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**